src/Runner/FileLintingIterator.php

```diff
@@ -20,7 +20,7 @@
  *
  * @internal
  */
-final class FileLintingIterator extends \IteratorIterator
+final class FileLintingIterator extends \CachingIterator
```

Comment:
so, let me ask same question here as well.
could you please describe what is a benefit of using `CachingIterator` in code you proposed ?

Reply:
Both iterators then provide the same iteration. w/o that one, the FileLinitingIterator has a different order than the other one. I can not say 100% for sure why that is, but I debugged it the last three days and this change does make that integration test green w/ PHP 5.6 (process linter) and PHP 7.2 (the other linter). Sorry for not being able to share more on this, it's just I can't say. But feel free to ask more, I've looked in there extensively the last days.

Reply:
order of items in iteration is not stable regardless which iterator you gonna use. basically when we read the files from drive, we could get different order each time.
I can see no benefit of this change, but I see CPU power wasted. Please revert it.

Reply:
Both of the decorating iterators are feed w/ the same order of files. So for a reason the caching iterator changes the order of current elements but I do not understand why that happens. Do you have any idea? It's not the order of files returned from disk, it is the same as far I've seen.

Reply:
order of iterating is not defined as stable anywhere, and it doesn't need to be stable tbh.

again, to have this PR merged, revert this change.
if you want this change to happen, send it as separated PR with dedicated utest that fails without change and works with it.

Reply:
Yes, we can separate this problem, I'll refresh the PR. Just wondering, what is a utest? Alternatively the fixture could be changed to have the order undefined. Should work as well (will make the test a little bit more convoluted, but likely not much compared to its current state).

Reply:
utest - unit test
fixture and undefined order - it shall be done indeed, order of reading files from drive is not defined, as we don't sort files after read from drive.